backend/src/main/java/com/benine/backend/http/CameraInfoHandler.java

```diff
@@ -1,24 +1,124 @@
 package com.benine.backend.http;
 
-import com.benine.backend.LogEvent;
-import com.sun.net.httpserver.HttpExchange;
+import com.benine.backend.camera.Camera;
+import com.benine.backend.camera.FocussingCamera;
+import com.benine.backend.camera.IrisCamera;
+import com.benine.backend.camera.MovingCamera;
+import com.benine.backend.camera.ZoomingCamera;
+import com.benine.backend.video.StreamType;
+import org.eclipse.jetty.server.Request;
 
 import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
-/**
- * Handles requests from the client about which camera's there are.
- * Created on 4-5-16.
- */
-public class CameraInfoHandler extends RequestHandler {
+public class CameraInfoHandler extends CameraRequestHandler {
+
+  private CameraStreamHandler streamHandler;
+
+  private CameraFocusHandler focusHandler;
+
+  private CameraMovingHandler moveHandler;
+
+  private CameraIrisHandler irisHandler;
+
+  private CameraZoomHandler zoomHandler;
+
+  /**
+   * Constructor for a new CameraInfoHandler, handling the /camera/ request.
+   */
+  public CameraInfoHandler() {
+    this.streamHandler = new CameraStreamHandler();
+    this.focusHandler = new CameraFocusHandler();
+    this.irisHandler = new CameraIrisHandler();
+    this.moveHandler = new CameraMovingHandler();
+    this.zoomHandler = new CameraZoomHandler();
+  }
+
+  @Override
+  public void handle(String s, Request request, HttpServletRequest req, HttpServletResponse res)
+          throws IOException, ServletException {
+
+    String cameraInfo = getCameraController().getCamerasJSON();
+
+    boolean routed = false;
+
+    if (checkValidCameraID(request)) {
+      String route = getRoute(request);
+      Camera cam = getCameraController().getCameraById(getCameraId(request));
+
+      switch (route) {
```

Comment:
I see lots of duplicate code here. Wouldn't it be better to store a string in each handler, create a local var handler and then loop over the list of handlers setting the local var handler to the handler with the string matching to route and then handle the request with that handler.